core/input/input_event.cpp

```diff
@@ -1725,22 +1725,22 @@ String InputEventMIDI::to_string() {
 	String ret;
 	switch (message) {
 		case MIDIMessage::NOTE_ON:
-			ret = vformat("Note On: channel=%d, pitch=%d, velocity=%d", channel, pitch, velocity);
+			ret = vformat("Note On: channel=%d, pitch=%d, velocity=%d, device=%d", channel, pitch, velocity, get_device());
 			break;
 		case MIDIMessage::NOTE_OFF:
-			ret = vformat("Note Off: channel=%d, pitch=%d, velocity=%d", channel, pitch, velocity);
+			ret = vformat("Note Off: channel=%d, pitch=%d, velocity=%d, device=%d", channel, pitch, velocity, get_device());
 			break;
 		case MIDIMessage::PITCH_BEND:
-			ret = vformat("Pitch Bend: channel=%d, pitch=%d", channel, pitch);
+			ret = vformat("Pitch Bend: channel=%d, pitch=%d, device=%d", channel, pitch, get_device());
 			break;
 		case MIDIMessage::CHANNEL_PRESSURE:
-			ret = vformat("Channel Pressure: channel=%d, pressure=%d", channel, pressure);
+			ret = vformat("Channel Pressure: channel=%d, pressure=%d, device=%d", channel, pressure, get_device());
 			break;
 		case MIDIMessage::CONTROL_CHANGE:
-			ret = vformat("Control Change: channel=%d, controller_number=%d, controller_value=%d", channel, controller_number, controller_value);
+			ret = vformat("Control Change: channel=%d, controller_number=%d, controller_value=%d, device=%d", channel, controller_number, controller_value, get_device());
 			break;
 		default:
-			ret = vformat("channel=%d, message=%d, pitch=%d, velocity=%d, pressure=%d, controller_number=%d, controller_value=%d, instrument=%d", channel, message, pitch, velocity, pressure, controller_number, controller_value, instrument);
+			ret = vformat("channel=%d, message=%d, pitch=%d, velocity=%d, pressure=%d, controller_number=%d, controller_value=%d, instrument=%d, device=%d", channel, message, pitch, velocity, pressure, controller_number, controller_value, instrument, get_device());
```

Comment:
None of the other `InputEvent::to_string` functions display their device id. To be consistent with the other input events, I would prefer, if it wouldn't be included.
What is your motivation to include the device id in `InputEventMIDI::to_string`?

Reply:
Its a Idiosyncrasy of being a music instrument input, i guess.
Not the author, but it makes sense to have the instrument along with the notes.

But... I agree with having consistency on the implementation, and to simple use the ".device" attribute to infer the origin of message, and by match one can use the to_string.

Reply:
since the vast majority of input events in games come from keyboard/mouse, and (at least before I finish the other branch I'm working on - which does the same as this one, but for mouse and keyboard) they get merged by the OS and therefore not being relevant, for now this leaves only gamepads with relevant `device`. So my understanding is device wasn't included in the string because it wasn't relevant. 

If youse feel this will clutter the output I can obviously remove it.

Reply:
Removed!